plugins/offers_invreq_hook.c

```diff
@@ -476,10 +468,36 @@ static struct command_result *invreq_base_amount_simple(struct command *cmd,
 
 static struct command_result *handle_amount_and_recurrence(struct command *cmd,
 							   struct invreq *ir,
-							   struct amount_msat amount)
+							   struct amount_msat base_inv_amount)
 {
+	/* BOLT-offers #12:
+	 * - if the offer included `amount`:
+	 *...
+	 *   - if the request contains `amount`:
+	 *     - MUST fail the request if its `amount` is less than the *base invoice amount*.
+	 *     - MAY fail the request if its `amount` is much greater than the *base invoice
+	 *     amount*.
+	 *     - MUST use the request's `amount` as the *base invoice amount*.
+	 */
+	if (ir->offer->amount && ir->invreq->amount) {
+		if (amount_msat_less(amount_msat(*ir->invreq->amount), base_inv_amount)) {
+			return fail_invreq(cmd, ir, "Amount must be less than %s",
+					   type_to_string(tmpctx, struct amount_msat,
+							  &base_inv_amount));
+		}
+		/* Easier to divide by 5 and compare, than multiply by 5 */
+		if (amount_msat_greater(amount_msat_div(amount_msat(*ir->invreq->amount), 5),
+					base_inv_amount)) {
+			return fail_invreq(cmd, ir, "Amount vastly exceeds %s",
+					   type_to_string(tmpctx, struct amount_msat,
+							  &base_inv_amount));
+		}
```

Comment:
Feels a bit like a magic limitation. Is there a BOLT-quote to associate with this?

Reply:
Yes, it's deliberately vague, but I'll add it.